backend/kesaseteli/applications/jobs/daily/tests/test_clean_youth_applications.py

```diff
@@ -0,0 +1,132 @@
+
+    # Active applications should haven't been cleaned
+    for application in active_applications_created_week_ago:
+        assert YouthApplication.objects.filter(pk=application.id).exists()
+
+
+@pytest.mark.django_db
+def test_clean_rejected_youth_applications_older_than_year():
+    date_year_ago = timezone.now() - relativedelta(year=1)
+    date_less_than_year_ago = timezone.now() - relativedelta(years=1, days=-1)
+
+    # Create different applications
+    rejected_applications_created_year_ago = setCreationDate(
+        RejectedYouthApplicationFactory.create_batch(size=5), created_at=date_year_ago
+    )
+
+    rejected_applications_created_less_than_year_ago = setCreationDate(
+        RejectedYouthApplicationFactory.create_batch(size=5),
+        created_at=date_less_than_year_ago,
+    )
+
+    unhandled_applications_created_year_ago = setCreationDate(
+        RejectableYouthApplicationFactory.create_batch(size=5), created_at=date_year_ago
+    )
+
+    accepted_applications_created_year_ago = setCreationDate(
+        AcceptedYouthApplicationFactory.create_batch(size=5), created_at=date_year_ago
+    )
+
+    # Run the batch job
+    clean_rejected_youth_applications.Job().execute()
+
+    # Rejected application older than year should have been cleaned
+    for application in rejected_applications_created_year_ago:
+        assert not YouthApplication.objects.filter(pk=application.id).exists()
+
+    # Newer rejected applications should haven't been cleaned
+    for application in rejected_applications_created_less_than_year_ago:
+        assert YouthApplication.objects.filter(pk=application.id).exists()
+
+    # Unhandled applications should haven't been cleaned
+    for application in unhandled_applications_created_year_ago:
+        assert YouthApplication.objects.filter(pk=application.id).exists()
+
+    # Accepted applications should haven't been cleaned
+    for application in accepted_applications_created_year_ago:
+        assert YouthApplication.objects.filter(pk=application.id).exists()
+
+
+@pytest.mark.django_db
+def test_clean_all_youth_applications_older_than_five_years():
+    date_five_years_ago = timezone.now() - relativedelta(years=5)
+    date_under_five_years_ago = timezone.now() - relativedelta(years=5, days=-1)
+
+    # Create different applications
+    applications_created_five_years_ago = setCreationDate(
+        YouthApplicationFactory.create_batch(size=20),
+        created_at=date_five_years_ago,
+    )
+
+    applications_created_less_than_five_years_ago = setCreationDate(
+        YouthApplicationFactory.create_batch(size=20),
+        created_at=date_under_five_years_ago,
+    )
+
+    # Run the batch job
+    clean_old_youth_applications.Job().execute()
+
+    # Old applications should have been cleaned
+    for application in applications_created_five_years_ago:
+        assert not YouthApplication.objects.filter(pk=application.id).exists()
+
+    # Newer applications should haven't been cleaned
```

Comment:
["should haven't" -> "shouldn't have" / "should not have"](https://ell.stackexchange.com/questions/156602/shouldnt-have-been-or-should-havent-been)

Reply:
Maybe without negation: `Should still exist` is the best